File: FrontEnd/src/app/models/Status.ts
export enum Status {
  DRAFT = 0,
  INCOMPLETE = 1,
  ACTIVE = 2,
  DEPRECATED = 3
}

export class StatusHelper {
  public static getName(status: Status): string {
    switch (status) {
      case Status.DRAFT: return 'Draft';
      case Status.INCOMPLETE: return 'Incomplete';
      case Status.ACTIVE: return 'Active';
      case Status.DEPRECATED: return 'Deprecated';
      default: return 'Unknown ['+status+']';
    }
  }
}

File: common/src/main/java/org/endeavourhealth/informationmanager/common/models/Status.java
package org.endeavourhealth.informationmanager.common.models;

import com.fasterxml.jackson.annotation.JsonValue;

public enum Status {
    DRAFT ((short)0, "Draft"),
    INCOMPLETE ((short)1, "Incomplete"),
    ACTIVE ((short)2, "Active"),
    DEPRECATED ((short)3, "Deprecated");

    private final Short value;
    private final String name;
    Status(short value, String name) {
        this.value = value;
        this.name = name;
    }

    @JsonValue
    public short getValue() { return this.value; }
    public String getName() { return this.name; }
    public String toString() {
        return this.value.toString();
    }

    public static Status byValue(Short value) {
        for (Status t: Status.values()) {
            if (t.value.equals(value))
                return t;
        }

        return null;
    }
}
